src/main/java/com/commercetools/project/sync/Syncer.java

```diff
@@ -47,38 +53,66 @@
   private final SphereClient sourceClient;
   private final SphereClient targetClient;
-  private final B sync;
-  private final C query;
+  private final CustomObjectService customObjectService;
 
+  /**
+   * Instantiates a {@link Syncer} which is used to sync resources from a source to a target
+   * commercetools project.
+   *
+   * @param sync The sync module that is used for syncing the transformed drafts to the target
+   *     project.
+   * @param sourceClient the client used for querying data from the source commercetools project.
+   * @param targetClient the client used for syncing the transformed drafts into the target
+   *     commercetools project.
+   * @param customObjectService service that is used for fetching and persisting the last sync
+   *     timestamp for delta syncing.
+   */
   public Syncer(
       @Nonnull final B sync,
-      @Nonnull final C query,
       @Nonnull final SphereClient sourceClient,
-      @Nonnull final SphereClient targetClient) {
+      @Nonnull final SphereClient targetClient,
+      @Nonnull final CustomObjectService customObjectService) {
     this.sync = sync;
-    this.query = query;
     this.sourceClient = sourceClient;
     this.targetClient = targetClient;
+    this.customObjectService = customObjectService;
   }
 
   /**
-   * Fetches the {@code CTP_SOURCE_CLIENT} project resources of type {@code T} with all needed
-   * references expanded and treats each page as a batch to the sync process. Then executes the sync
-   * process of all pages in parallel. It then returns a completion stage containing no result after
-   * the execution of the sync process and logging the result.
+   * Fetches the sourceClient's project resources of type {@code T} with all needed references
+   * expanded and treats each page as a batch to the sync process. Then executes the sync process of
+   * all pages in parallel. It then returns a completion stage containing no result after the
+   * execution of the sync process and logging the result.
+   *
+   * <p>Note: The method checks if there was a last sync time stamp persisted as a custom object in
+   * the target project for this specific source project and sync module. If there is, it will sync
+   * only the resources which were modified after the last sync time stamp and before the start of
+   * this sync.
    *
    * @return completion stage containing no result after the execution of the sync process and
    *     logging the result.
    */
   public CompletionStage<Void> sync() {
 
+    final String sourceProjectKey = sourceClient.getConfig().getProjectKey();
+    final String syncModuleName = getSyncModuleName(sync.getClass());
     if (LOGGER.isInfoEnabled()) {
-      LOGGER.info(format("Starting %s..", sync.getClass().getSimpleName()));
+      final String targetProjectKey = targetClient.getConfig().getProjectKey();
+      LOGGER.info(
+          format(
+              "Starting %s from CTP project with key '%s' to project with key '%s'",
```

Comment:
hmm, I know it's a CTP project, I think simple is better.  Starting sync from `a` to `b`. wdyt ?

Reply:
I didn't want to confuse the user and put them into the direction of thinking what is it the project name or key. Don't you think explicitly stating is better? I mean how does this make it not simple or complex for the user?

Reply:
hmm true, you are right 👍